src/main/java/com/commercetools/project/sync/customer/CustomerSyncer.java

```diff
@@ -0,0 +1,95 @@
+import com.commercetools.sync.customers.CustomerSync;
+import com.commercetools.sync.customers.CustomerSyncOptions;
+import com.commercetools.sync.customers.CustomerSyncOptionsBuilder;
+import com.commercetools.sync.customers.helpers.CustomerSyncStatistics;
+import com.commercetools.sync.customers.utils.CustomerReferenceResolutionUtils;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.customers.Customer;
+import io.sphere.sdk.customers.CustomerDraft;
+import io.sphere.sdk.customers.queries.CustomerQuery;
+import java.time.Clock;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CustomerSyncer
+    extends Syncer<
+        Customer,
+        CustomerDraft,
+        CustomerSyncStatistics,
+        CustomerSyncOptions,
+        CustomerQuery,
+        CustomerSync> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CustomerSyncer.class);
+
+  private CustomerSyncer(
+      @Nonnull final CustomerSync customerSync,
+      @Nonnull final SphereClient sourceClient,
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final CustomObjectService customObjectService,
+      @Nonnull final Clock clock) {
+    super(customerSync, sourceClient, targetClient, customObjectService, clock);
+  }
+
+  public static CustomerSyncer of(
+      @Nonnull final SphereClient sourceClient,
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final Clock clock) {
+    final QuadConsumer<
+            SyncException,
+            Optional<CustomerDraft>,
+            Optional<Customer>,
+            List<UpdateAction<Customer>>>
+        logErrorCallback =
+            (exception, newResourceDraft, oldResource, updateActions) ->
+                logErrorCallback(
+                    LOGGER,
+                    "customer",
+                    exception,
+                    oldResource.map(Customer::getEmail).orElse(""),
+                    updateActions);
+    final TriConsumer<SyncException, Optional<CustomerDraft>, Optional<Customer>>
+        logWarningCallback =
+            (exception, newResourceDraft, oldResource) ->
+                logWarningCallback(
+                    LOGGER, "customer", exception, oldResource.map(Customer::getEmail).orElse(""));
+    final CustomerSyncOptions customerSyncOptions =
+        CustomerSyncOptionsBuilder.of(targetClient)
+            .errorCallback(logErrorCallback)
+            .warningCallback(logWarningCallback)
+            .build();
+    CustomerSync customerSync = new CustomerSync(customerSyncOptions);
+    final CustomObjectService customObjectService = new CustomObjectServiceImpl(targetClient);
+    return new CustomerSyncer(customerSync, sourceClient, targetClient, customObjectService, clock);
```

Comment:
I would leave an empty line before this line to highlight we are creating the Syncer and returning that object. Just like other existing Syncer classes (CategorySyncer)

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/184/commits/adf0a9e3907e884e3fed80aa76b76b210346d69f